src/BlockTypes/ProductQuery.php

```diff
@@ -814,4 +816,61 @@ function( $rating ) use ( $product_visibility_terms ) {
 		);
 	}
 
+
+	/**
+	 * Return a query to filter products by taxonomies (product categories, product tags, etc.)
+	 *
+	 * For example:
+	 * User could provide "Product Categories" using "Filters" ToolsPanel available in Inspector Controls.
+	 * We use this function to extract it's query from $tax_query.
+	 *
+	 * For example, this is how the query for product categories will look like in $tax_query array:
+	 * Array
+	 *    (
+	 *        [taxonomy] => product_cat
+	 *        [terms] => Array
+	 *            (
+	 *                [0] => 36
+	 *            )
+	 *    )
+	 *
+	 * For product categories, taxonomy would be "product_tag"
+	 *
+	 * @param array $tax_query taxonomy query.
+	 * @return array Query to filter products by taxonomies.
+	 */
+	private function get_filter_by_taxonomies_query( $tax_query ): array {
+		if ( ! is_array( $tax_query ) ) {
+			return [];
+		}
+
+		/**
+		 * Get an array of taxonomy names associated with the "product" post type because
+		 * we also want to include custom taxonomies associated with the "product" post type.
+		 */
+		$product_taxonomies = get_taxonomies( array( 'object_type' => array( 'product' ) ), 'names' );
+		$result             = array_filter(
+			$tax_query,
+			function( $item ) use ( $product_taxonomies ) {
+				return isset( $item['taxonomy'] ) && in_array( $item['taxonomy'], $product_taxonomies, true );
+			}
+		);
+
+		return ! empty( $result ) ? [ 'tax_query' => $result ] : [];
+	}
+
+	/**
+	 * Returns the keyword filter from the given query.
+	 *
+	 * @param WP_Query $query The query to extract the keyword filter from.
+	 * @return array The keyword filter, or an empty array if none is found.
+	 * @throws InvalidArgumentException If $query is not an array.
```

Comment:
I thank that we should do this check, in this way, we will not throw any errors. Now we don't catch the error.